### PR TITLE
Add a SQLite query planner for type detection

### PR DESCRIPTION
**Pull request by theopolis**
At this point, this is a proof of concept solution for #1649.

### CODE REVIEW

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

File: CMake/FindSqlite3.cmake [hunk 1]
```diff
@@ -1,2 +1,15 @@
+add_definitions(
+  -DSQLITE_ENABLE_COLUMN_METADATA
+  -DSQLITE_ENABLE_API_ARMOR
+  # Functionality disables
+  -DSQLITE_DISABLE_LFS
```
Comment (theopolis) [inline CMake/FindSqlite3.cmake]: These generate more unused variable warnings within the SQLite code, harmless, but annoying. :/

File: osquery/sql/sqlite_util.cpp [hunk 2]
```diff
@@ -160,6 +212,52 @@ SQLiteDBManager::~SQLiteDBManager() {
   }
 }
 
+QueryPlanner::QueryPlanner(const std::string& query, sqlite3* db) {
+  queryInternal("EXPLAIN QUERY PLAN " + query, plan_, db);
+  queryInternal("EXPLAIN " + query, program_, db);
+
+  for (const auto& row : plan_) {
+    auto details = osquery::split(row.at("detail"));
+    tables_.push_back(details[2]);
```
Comment (theopolis) [inline osquery/sql/sqlite_util.cpp]: If the tables are deduced here, we do not need to keep the `QUERY PLAN` results around as a member variable.

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by theopolis** [comment]
I forgot to switch the test_main when I did the rebase ;)

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Pull request merged**